libs/pages/services/src/lib/feature/page-job-create-feature/template-form-context-sync.tsx

```diff
@@ -0,0 +1,115 @@
+import { APIVariableScopeEnum, type JobLifecycleTypeEnum } from 'qovery-typescript-axios'
+import { type ReactNode, useEffect, useState } from 'react'
+import { useLifecycleTemplate } from '@qovery/domains/environments/feature'
+import { type ServiceTemplateOptionType } from '../page-new-feature/service-templates'
+import { useJobContainerCreateContext } from './page-job-create-feature'
+
+function getLifecycleType(option?: string): JobLifecycleTypeEnum {
+  if (option?.includes('terraform')) {
+    return 'TERRAFORM'
+  }
+
+  if (option?.includes('cloudformation')) {
+    return 'CLOUDFORMATION'
+  }
+
+  return 'GENERIC'
+}
+
+interface TemplateFormContextSyncProps {
+  environmentId: string
+  templateData: ServiceTemplateOptionType & { template_id: string }
+  children: ReactNode
+}
+
+export function TemplateFormContextSync({
```

Comment:
I wonder if `context` is a good word because it can refer to to `context` feature in React. Perhaps `TemplateFormDataSync` is more coherent ?

Reply:
Renamed to `TemplateFormSync`